test/e2e/stack_test.go

```diff
@@ -33,8 +33,8 @@ import (
 // TestVersionUpgradeOrdering deploys the entire stack, with resources associated together.
 // Then, it updates their version, and ensures a strict ordering is respected during the version upgrade.
 func TestVersionUpgradeOrdering(t *testing.T) {
```

Comment:
Nit: maybe rename the test?
```suggestion
func TestMajorVersionUpgradeOrdering(t *testing.T) {
```